backend/src/cache/main.go

```diff
@@ -101,3 +118,20 @@ func main() {
 	}
 	log.Fatal(server.ListenAndServeTLS(certPath, keyPath))
 }
+
+func parseS3Flags() S3Params {
+	var params S3Params
+	flag.StringVar(&params.serviceHost, "s3_service_host", os.Getenv("MINIO_SERVICE_SERVICE_HOST"), "hostname of S3.")
+	flag.StringVar(&params.servicePort, "s3_service_port", os.Getenv("MINIO_SERVICE_SERVICE_PORT"), "port of S3.")
+	flag.StringVar(&params.region, "s3_region", os.Getenv("MINIO_SERVICE_REGION"), "Region of S3 service.")
+	flag.StringVar(&params.accessKey, "s3_access_key", os.Getenv("OBJECTSTORECONFIG_ACCESSKEY"), "accessKey of S3.")
+	flag.StringVar(&params.secretKey, "s3_secret_key", os.Getenv("OBJECTSTORECONFIG_SECRETACCESSKEY"), "secretKey of S3.")
+	flag.BoolVar(&params.isServiceSecure, "s3_is_service_secure",
+		common.GetBoolFromStringWithDefault(os.Getenv("MINIO_SERVICE_SECURE"), false), "is ssl enabled on S3.")
+	flag.StringVar(&params.bucketName, "s3_bucketname", os.Getenv("MINIO_PIPELINE_BUCKET_NAME"), "default bucketname on S3.")
```

Comment:
PIPELINE_BUCKET_NAME, SERVICE_HOST and SERVICE_SECURE is more generic. You are mixing it quite heavily. Sometimes  the prefix is OBJECTSTORECONFIG and sometimes MINIO_SERVICE. Please make it consistent within and also with the go functions above

Reply:
Please have a look at https://github.com/kubeflow/manifests/blob/master/apps/pipeline/upstream/base/cache/cache-deployment.yaml and https://github.com/kubeflow/pipelines/blob/master/manifests/kustomize/base/cache/cache-deployment.yaml for consistency

Reply:
i checked how apiserver uses the envs -> https://github.com/kubeflow/pipelines/blob/74c7773ca40decfd0d4ed40dc93a6af591bbc190/backend/src/apiserver/client_manager.go#L38

and how its deployed -> https://github.com/kubeflow/pipelines/blob/master/manifests/gcp_marketplace/chart/kubeflow-pipelines/templates/pipeline.yaml#L788 naming is also different here.

Reply:
but anyway i will use the suggested names